rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt

```diff
@@ -121,7 +128,7 @@ data class IntValue(val value: Long) : NumberValue, Value() {
 
     override fun asInt() = this
     // TODO Verify conversion
-    override fun asDecimal(): DecimalValue = DecimalValue(BigDecimal(value))
+    override fun asDecimal(): DecimalValue = DecimalValue(internalValue)
```

Comment:
I would add a blank line before the comment. Also, I do not undersand what needs to be verified here.
Finally, in this case you can remove the return type of the method as it is obvious